pkg/apis/core/types.go

```diff
@@ -3683,7 +3683,9 @@ type NodeSpec struct {
 	// +optional
 	ProviderID string
 
-	// Unschedulable controls node schedulability of new pods. By default node is schedulable.
+	// Deprecated. Unschedulable controls node schedulability of new pods. By default, node is schedulable.
+	// Remove field after 1.21.
```

Comment:
we do not remove fields from a given version, see links to the deprecation policy

Reply:
So, we can't deprecate the API unless we have a v2. I noticed the `ExternalID` was marked as deprecated and prefixed with `DoNotUse`, can we do the same thing to `Unschedulable` field here?

```go
	// Deprecated. Not all kubelets will set this field. Remove field after 1.13.
	// see: https://issues.k8s.io/61966
	// +optional
	DoNotUseExternalID string `json:"externalID,omitempty" protobuf:"bytes,2,opt,name=externalID"`
```

BTW: Do we have plans or documents on when we could start with a new major version of API?

Reply:
> So, we can't deprecate the API unless we have a v2.

Correct

> I noticed the ExternalID was marked as deprecated and prefixed with DoNotUse, can we do the same thing to Unschedulable field here?

The use case for ExternalID was much more narrow and had a single writer (the node). I would not change the name of the Unschedulable field.

> Do we have plans or documents on when we could start with a new major version of API?

Not currently

Reply:
OK, I'll close this PR since we can't do this for now.

/close